From 56aff665cb0100a01e32d7dd3ed03deb957a1a49 Mon Sep 17 00:00:00 2001
From: Muhammad <muhammad23012009@hotmail.com>
Date: Sun, 11 Aug 2024 16:40:16 +0500
Subject: [PATCH 12/12] (halium) drop android.hardware.power-ndk_export_shared
 * Link with android.hardware.power-V5-ndk ourselves * Also link with
 android.hardware.common{.fmq}

Change-Id: I10eadefc12d842ae1f56ae0026c729c244c5baa3
Signed-off-by: Muhammad <muhammad23012009@hotmail.com>
---
 services/powermanager/Android.bp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/services/powermanager/Android.bp b/services/powermanager/Android.bp
index f7c8a81..d2e607d 100644
--- a/services/powermanager/Android.bp
+++ b/services/powermanager/Android.bp
@@ -9,7 +9,6 @@ package {
 
 cc_library_shared {
     name: "libpowermanager",
-    defaults: ["android.hardware.power-ndk_export_shared"],
     srcs: [
         "BatterySaverPolicyConfig.cpp",
         "CoolingDevice.cpp",
@@ -46,13 +45,17 @@ cc_library_shared {
         "android.hardware.power@1.1",
         "android.hardware.power@1.2",
         "android.hardware.power@1.3",
+        "android.hardware.common-V2-ndk",
+        "android.hardware.common.fmq-V1-ndk",
+        "android.hardware.power-V5-ndk",
     ],
 
-    export_shared_lib_headers: [
+    export_static_lib_headers: [
         "android.hardware.power@1.0",
         "android.hardware.power@1.1",
         "android.hardware.power@1.2",
         "android.hardware.power@1.3",
+        "android.hardware.power-V5-ndk",
     ],
 
     cflags: [
-- 
2.45.2

